asnake/aspace/__init__.py

```diff
@@ -0,0 +1,138 @@
+from asnake.client import ASnakeClient
+import json
+
+class ASpace():
+
+	# this happens when you call ASpace()
+	def __init__(self, repository=None):
+		
+		# Repository will default to 2 if not provided
```

Comment:
This is complicated by the fact that repositories don't remotely have all of their sub-objects on them as properties when fetched - I think we're going to need to special-case this in some wise.

Naive guess - process the schemas from ArchivesSpace, use uri to determine what's nested under resource _in practice_, and have getattr for repositories special-case to a merged dict of `properties on repo obj + known repo-scoped things.'

I think that might be ok for a first pass, and if it doesn't get close enough to full functionality for fetching stuff, we can give up and go to something more explicitly "we have to write a bunch of explicit code to handle all methods."

Reply:
I would have assumed to call another ASpace() to use another repository, either directly or in a loop through repositories, I guess the drawback would be you have to call an ASpace(), loop to get repo ids and then call a new ASpace() for each. That's what I would except, but if there's a better way we can try to change it.

Reply:
Yeah, I definitely think we want to avoid that if at all possible.  I think if we move repository-scoped methods down to repository (which can either be a custom object with a merger of repository properties and repo-scoped methods) and have methods at the ASpace() level operate over all repositories per: https://github.com/archivesspace-labs/ArchivesSnake/blob/greg/asnake/aspace/__init__.py#L50, that'll be coherent and work like people expect?